modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/websocket/WebSocketAPITopicsTestCase.java

```diff
@@ -0,0 +1,359 @@
+import org.wso2.am.integration.clients.store.api.v1.dto.ApplicationDTO;
+import org.wso2.am.integration.clients.store.api.v1.dto.ApplicationKeyDTO;
+import org.wso2.am.integration.clients.store.api.v1.dto.ApplicationKeyGenerateRequestDTO;
+import org.wso2.am.integration.clients.store.api.v1.dto.SubscriptionDTO;
+import org.wso2.am.integration.test.utils.APIManagerIntegrationTestException;
+import org.wso2.am.integration.test.utils.base.APIMIntegrationBaseTest;
+import org.wso2.am.integration.test.utils.base.APIMIntegrationConstants;
+import org.wso2.am.integration.test.utils.bean.APILifeCycleAction;
+import org.wso2.am.integration.test.utils.bean.APIRequest;
+import org.wso2.am.integration.test.utils.generic.APIMTestCaseUtils;
+import org.wso2.am.integration.tests.streamingapis.StreamingApiTestUtils;
+import org.wso2.am.integration.tests.websocket.client.WebSocketClientImpl;
+import org.wso2.am.integration.tests.websocket.server.topic.WebSocketRootTopicServlet;
+import org.wso2.am.integration.tests.websocket.server.topic.WebSocketTopicServlet;
+import org.wso2.carbon.apimgt.api.model.APIIdentifier;
+import org.wso2.carbon.automation.engine.annotations.ExecutionEnvironment;
+import org.wso2.carbon.automation.engine.annotations.SetEnvironment;
+import org.wso2.carbon.automation.engine.context.TestUserMode;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+import org.wso2.carbon.integration.common.utils.mgt.ServerConfigurationManager;
+import org.wso2.carbon.utils.xml.StringUtils;
+
+import java.net.InetAddress;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import static org.testng.Assert.assertTrue;
+
+@SetEnvironment(executionEnvironments = {ExecutionEnvironment.STANDALONE})
+public class WebSocketAPITopicsTestCase extends APIMIntegrationBaseTest {
+
+    private final Log log = LogFactory.getLog(WebSocketAPITopicsTestCase.class);
+    enum AUTH_IN {
+        HEADER,
+        QUERY
+    }
+    private final ExecutorService executorService = Executors.newSingleThreadExecutor();
+    private final String apiName = "WebSocketAPI";
+    private final String applicationName = "WebSocketApplication";
+    private String apiEndPoint;
+    private String provider;
+    private APIRequest apiRequest;
+    private int webSocketServerPort;
+    private String webSocketServerHost;
+    private ServerConfigurationManager serverConfigurationManager;
+    private String websocketAPIID;
+    String appId;
+
+    @Factory(dataProvider = "userModeDataProvider")
+    public WebSocketAPITopicsTestCase(TestUserMode userMode) {
+
+        this.userMode = userMode;
+    }
+
+    @DataProvider
+    public static Object[][] userModeDataProvider() {
+
+        return new Object[][]{
+                new Object[] { TestUserMode.SUPER_TENANT_ADMIN },
+                new Object[] { TestUserMode.TENANT_ADMIN }
```

Comment:
Don't we need to run this for other user modes? Eg: email user, secondary user store, tenant user etc

Reply:
We decided these two are enough for Streaming APIs in an earlier offline discussion with @tharindu1st . Websub[1] and SSE [2] streaming API test cases also have been written in this way.
[1] https://github.com/wso2/product-apim/blob/master/modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/streamingapis/websub/WebSubAPITestCase.java#L118
[2] https://github.com/wso2/product-apim/blob/master/modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/am/integration/tests/streamingapis/serversentevents/ServerSentEventsAPITestCase.java#L125